src/clients/nextjs.ts

```diff
@@ -0,0 +1,165 @@
+      | false
+      | {
+          [param: string]: any;
+        };
+  }>(),
+  setValue: async () => {
+    try {
+      const pagesManifest = (await loadManifest(
+        join(DOT_NEXT, PAGES_MANIFEST),
+        false
+      )) as Record<string, string>;
+      const appPathsManifest = (await loadManifest(
+        join(DOT_NEXT, APP_PATHS_MANIFEST),
+        false
+      )) as Record<string, string>;
+
+      const appPathsKeys = Object.keys(appPathsManifest);
+      for (let i = 0; i < appPathsKeys.length; i++) {
+        const key = appPathsKeys[i];
+        const path = key.replace(
+          /\/(page|not-found|layout|loading|head|route)$/,
+          ''
+        );
+        const reg = getRouteRegex(path);
+        const matcher = getRouteMatcher(reg);
+
+        PATHS_CACHE.value.add({
+          route: path,
+          matcher,
+          re: reg.re
+        });
+      }
+
+      const keys = Object.keys(pagesManifest);
+      for (let i = 0; i < keys.length; i++) {
+        const key = keys[i];
+        const reg = getRouteRegex(key);
+        const matcher = getRouteMatcher(reg);
+
+        PATHS_CACHE.value.add({
+          route: key,
+          matcher,
+          re: reg.re
+        });
+      }
+    } catch (e) {}
+  },
+  keepUpdated: () => {
+    const watcher = chokidar.watch(DOT_NEXT, {
+      ignoreInitial: true
+    });
+
+    watcher.on('all', () => {
+      PATHS_CACHE.setValue();
+    });
+    return watcher;
+  }
+};
+
+const parsedPathname = (url: string) => {
+  return url.split('?')[0];
+};
+
+const wrappedHandler = (
+  handler: ReturnType<NextNodeServer['getRequestHandler']>,
+  ctx: {
+    getParameterizedRoute: (route: string) => string;
+    counter: Counter;
+    histogram: Histogram;
+  }
+) => {
+  return async function (
+    ...args: Parameters<ReturnType<NextNodeServer['getRequestHandler']>>
+  ) {
+    const [req, res] = args;
+    const start = process.hrtime.bigint();
+    const result = handler(...args);
+    if (result instanceof Promise) {
+      await result;
+    }
+    const end = process.hrtime.bigint();
+    const duration = Number(end - start) / 1e6;
+    const parsedPath = ctx.getParameterizedRoute(
+      parsedPathname(req.url ?? '/')
+    );
+
+    ctx.counter
+      .labels(
+        parsedPath !== '' ? parsedPath : '/',
+        req.method ?? 'GET',
+        res.statusCode?.toString() ?? '500'
+      )
+      .inc();
+
+    ctx.histogram
+      .labels(
+        parsedPath !== '' ? parsedPath : '/',
+        req.method ?? 'GET',
+        res.statusCode?.toString() ?? '500'
+      )
+      .observe(duration);
+
+    return result;
+  };
+};
+
+export const instrumentNextjs = (
+  nextServer: typeof NextNodeServer,
+  { counter, histogram }: { counter?: Counter; histogram?: Histogram }
+) => {
+  if (!counter || !histogram) {
+    throw new Error(
+      'counter and histogram are required. Make sure you are using openmetrics mode.'
+    );
+  }
```

Comment:
In which case this error will be raised, as we are passing these values ourselves.

Reply:
Yes we are passing the counter and histogram. But only for the `openmetrics` mode. I am aware we don't have otel changes out yet. But this is good to have. Can be removed.

Reply:
Let's remove it here and add with Otel change.